### PR TITLE
REF: Migrate to react navigation5

### PR DESCRIPTION
**Pull request by limpbrains**
I've tried to save the same Stacks structure.
Renamed `MainBottomTabs.js` to  `Navigation.js`

Still WIP
And of course these changes are devastating! Everything should be carefully retested

### CODE REVIEW

**Comment by ncoelho** [comment]
Massive! Gonna give it a test today and report back

**Comment by limpbrains** [comment]
> Massive! Gonna give it a test today and report back

Thanks to e2e test I've found some bugs, trying to fix them.

**Comment by limpbrains** [comment]
When navigating to send screen I'm not getting this warning:
https://reactnavigation.org/docs/troubleshooting/#i-get-the-warning-non-serializable-values-were-found-in-the-navigation-state

looks like passing Wallet or callbacks inside the navigator params it is a wrong pattern 
I think is ok for now, but probably show be fixed in future

**Comment by Overtorment** [comment]
@marcosrdz leave your thoughts when you  have a chance

**Comment by ncoelho** [comment]
Reporting back. No problems, bugs or errors found. And it was nice to see the new modals on the import file screen :)

**Comment by limpbrains** [comment]
I've tested deeplinks:
- long tap on app icon on ios
- paying btcpayserver invoice

Everything works fine. I believe it is ready

**Comment by Overtorment** [comment]
lets do a final round of tests before merging here https://github.com/BlueWallet/BlueWallet/pull/1154

**Comment by Overtorment** [comment]
@marcosrdz can you take a look? why is ios build failing?
maybe its related, I added `NODE_ENV=production` to our appcenter builds, so devDependencies dont get pulled, and fixed all issues (for master at least)


```
Loading dependency graph, done.
Failed to construct transformer:  { Error: Cannot find module 'metro-react-native-babel-preset/src/configs/hmr'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/Users/runner/runners/2.168.2/work/1/s/node_modules/metro-react-native-babel-transformer/src/index.js:81:23)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3) code: 'MODULE_NOT_FOUND' }
error Cannot read property 'transformFile' of undefined. Run CLI with --verbose flag for more details.
TypeError: Cannot read property 'transformFile' of undefined
    at /Users/runner/runners/2.168.2/work/1/s/node_modules/metro/src/Bundler.js:87:34
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/runner/runners/2.168.2/work/1/s/node_modules/metro/src/Bundler.js:14:24)
    at _next (/Users/runner/runners/2.168.2/work/1/s/node_modules/metro/src/Bundler.js:34:9)

##[error]Bash exited with code '1'.
```

File: package.json [hunk 1]
```diff
@@ -3,23 +3,23 @@
...
+    "babel-jest": "^26.0.1",
     "babel-preset-flow": "^6.23.0",
     "eslint": "^6.5.1",
     "eslint-plugin-babel": "^5.3.0",
     "eslint-plugin-import": "^2.18.0",
     "eslint-plugin-node": "^9.1.0",
     "eslint-plugin-promise": "^4.2.1",
     "eslint-plugin-react": "^7.14.2",
-    "flow-bin": "^0.102.0",
+    "flow-bin": "^0.125.1",
```
Comment (Overtorment) [inline package.json:19]: do we steel need flow? 
we are not using it

**Comment by Overtorment** [comment]
ok android buil fails as well with the same error.
something wring with deps..?

File: package.json [hunk 2]
```diff
@@ -3,23 +3,23 @@
...
     "jest": "^24.9.0",
     "jetifier": "^1.6.3",
-    "metro-react-native-babel-preset": "^0.56.0",
+    "metro-react-native-babel-preset": "^0.59.0",
```
Comment (Overtorment) [inline package.json:22]: can this be the issue? maybe it should move to prod dependencies? or maybe not even bumped at all (fair you need to bump certain deps only when you update RN itsel)

**Comment by Overtorment** [comment]
merged in https://github.com/BlueWallet/BlueWallet/pull/1154
thank you!
will cleanup some minor issues in master

**Comment by limpbrains** [comment]
> merged in #1154
> thank you!
> will cleanup some minor issues in master

HUGE SUCCESS
Thank you guys!

**Pull request merged**